mathics/core/read.py

```diff
@@ -316,3 +316,8 @@ def read_from_stream(stream, word_separators, msgfn, accepted=None):
                 break
 
             word += tmp
+
+
+def set_inputfile(filename):
+    global INPUTFILE_VAR
+    INPUTFILE_VAR = filename
```

Comment:
@adamantinum. Thanks for catching this and providing a solution. As a comment, a better place to store the `INPUT_FILE_NAME` variable would be as an attribute in the `Definitions` object.